healsparse/geom.py

```diff
@@ -0,0 +1,668 @@
+
+    if not isinstance(geom, (list, tuple)):
+        geom = [geom]
+
+    # split the geom objects up by value
+    gdict = {}
+    for g in geom:
+        value = g.value
+        _check_int(value)
+        _check_int_size(value, smap.dtype)
+
+        if value not in gdict:
+            gdict[value] = [g]
+        else:
+            gdict[value].append(g)
+
+    # deal with each value separately and add to
+    # the map
+    for value, glist in gdict.items():
+        for i, g in enumerate(glist):
+            tpixels = g.get_pixels(nside=smap.nsideSparse)
+            if i == 0:
+                pixels = tpixels
+            else:
+                oldsize = pixels.size
+                newsize = oldsize + tpixels.size
+                # need refcheck=False because it will fail when running
+                # the python profiler; I infer that the profiler holds
+                # a reference to the objects
+                pixels.resize(newsize, refcheck=False)
+                pixels[oldsize:] = tpixels
+
+        pixels = np.unique(pixels)
+
+        values = smap.getValuePixel(pixels)
+        values |= value
+        smap.updateValues(pixels, values)
+
+
+def _check_int(x):
+    check = (
+        issubclass(x.__class__, np.integer) or
+        issubclass(x.__class__, int)
+    )
+    if not check:
+        raise ValueError('value must be integer type, '
+                         'got %s' % x)
+
+
+def _check_int_size(value, dtype):
+    ii = np.iinfo(dtype)
+    if value < ii.min or value > ii.max:
+        raise ValueError('value %d outside range [%d, %d]' %
+                         (value, ii.min, ii.max))
+
+
+class GeomBase(object):
+    """
+    base class for goemetric objects that can convert
+    themselves to maps
+    """
+
+    @property
+    def value(self):
+        """
+        get the value to be used for all pixels in the map
+        """
+        return self._value
+
+    def get_pixels(self, *, nside):
+        """
+        get pixels for this map
+
+        Parameters
+        ----------
+        nside: int
+            Nside for the pixels
+        """
+        raise NotImplementedError('implment get_pixels')
+
+    def get_map(self, *, nside, dtype):
+        """
+        get a healsparse map corresponding to this geometric primitive
+
+        Parameters
+        ----------
+        size: integer
+            Size of the values array
+        dtype: np dtype
+            For the output array
+
+        Returns
+        -------
+        HealSparseMap
+        """
+
+        smap = HealSparseMap.makeEmpty(
+            nsideCoverage=32,
+            nsideSparse=nside,
+            dtype=dtype,
+            sentinel=0,
+        )
+        pixels = self.get_pixels(nside=nside)
+        values = self.get_values(size=pixels.size, dtype=dtype)
+        smap.updateValues(pixels, values)
+
+        return smap
+
+    def get_values(self, *, size, dtype):
+        """
```

Comment:
I guess this is an internal convenience function, should it be `_get_values()`?

Reply:
public